scripts/RecompileDelivery.php

```diff
@@ -0,0 +1,248 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2016 (original work) Open Assessment Technologies SA;
+ *
+ *
+ */
+namespace oat\taoDeliveryRdf\scripts;
+
+use oat\oatbox\action\Action;
+use common_report_Report as Report;
+use oat\taoDeliveryRdf\model\SimpleDeliveryFactory;
+use oat\oatbox\action\ResolutionException;
+
+//Load extension to define necessary constants.
+\common_ext_ExtensionsManager::singleton()->getExtensionById('taoTests');
+\common_ext_ExtensionsManager::singleton()->getExtensionById('taoDeliveryRdf');
+
+/**
+ * Class RecompileDelivery
+ * @package oat\taoDeliveryRdf\scripts
+ * @author Aleh Hutnikau, <hutnikau@1pt.com>
+ *
+ * Run example:
+ * ```
+ * sudo -u www-data php index.php 'oat\taoDeliveryRdf\scripts\RecompileDelivery'
```

Comment:
Indeed am unsure about how I should use the script
```
sudo -u www-data php index.php 'oat\taoDeliveryRdf\scripts\RecompileDelivery' compile http://tao.local/mytao.rdf#i14646778081995158
PHP Fatal error:  Call to a member function getUri() on null in /home/sam/Workspace/projects/TAO/PR/taoDeliveryRdf/scripts/RecompileDelivery.php on line 147
```
where http://tao.local/mytao.rdf#i14646778081995158 is a delivery I just created and got the uri from
```
sudo -u www-data php index.php 'oat\taoDeliveryRdf\scripts\RecompileDelivery' list
http://tao.local/mytao.rdf#i14649684368987171 - Delivery of Test 3
http://tao.local/mytao.rdf#i14646088199931108 - Delivery of QTI Example Test
http://tao.local/mytao.rdf#i14646754904861141 - Delivery of Test 2
http://tao.local/mytao.rdf#i14646778081995158 - Delivery of Test 3